### PR TITLE
EVAKA-4157 Daily service times UX enhancements & E2E tests

### PR DESCRIPTION
**Pull request by akheron**
#### Summary

- UX improvements:
    - Require at least one day to be selected in irregular times
    - Show irregular time inputs only after checking the checkbox for that day
    - Show time input errors only after they've been touched
    - Mobile: Indicate not set daily service times
- Add employee side e2e tests
- Reduce duplication in employee-side code

Screenshot-2021-04-15T13:17:08+03:00

### CODE REVIEW

File: frontend/src/e2e-playwright/pages/employee/child-information.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import { Page } from 'playwright'
+import { RawElement } from 'e2e-playwright/utils/element'
+
+export type Collapsible = 'dailyServiceTimes'
+
+export default class ChildInformationPage {
+  constructor(private readonly page: Page) {}
+
+  readonly #collapsibles: Record<Collapsible, RawElement> = {
```
Comment (Joosakur) [inline frontend/src/e2e-playwright/pages/employee/child-information.ts:9]: is this `#variable` some syntax I haven't seen before or some kind of naming convention?

File: frontend/src/e2e-playwright/specs/5_employee/child-information.spec.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    expect(await section.typeText).toEqual(
+      'Epäsäännöllinen varhaiskasvatusaika'
+    )
+    expect(await section.timesText).toEqual(
+      'maanantai 08:15-16:45, perjantai 09:00-13:30'
+    )
+  })
+})
+
+// fixture('Employee - Child Information')
```
Comment (Joosakur) [inline frontend/src/e2e-playwright/specs/5_employee/child-information.spec.ts:100]: should this commented code be removed?

File: frontend/src/e2e-playwright/pages/employee/child-information.ts (hunk 1 above)
Comment (akheron) [reply frontend/src/e2e-playwright/pages/employee/child-information.ts:9]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#private_field_declarations

It was used in some other page model, so I continued to use it.

File: frontend/src/employee-frontend/components/child-information/DailyServiceTimesSection.tsx [hunk 3]
```diff
@@ -195,6 +197,11 @@ const DailyServiceTimesSection = React.memo(function DailyServiceTimesSection({
           end:
             formData.type === 'REGULAR' ? required(formData.regular.end) : null
         },
+        irregular:
+          formData.type === 'IRREGULAR' &&
+          !weekdays.some((day) => formData[day].selected)
+            ? 'Valitse vähintään yksi päivä'
```
Comment (Joosakur) [inline frontend/src/employee-frontend/components/child-information/DailyServiceTimesSection.tsx:203]: could be moved into localization file

File: frontend/src/e2e-playwright/specs/5_employee/child-information.spec.ts (hunk 2 above)
Comment (akheron) [reply frontend/src/e2e-playwright/specs/5_employee/child-information.spec.ts]: Oops, removed

File: frontend/src/employee-frontend/components/child-information/DailyServiceTimesSection.tsx (hunk 3 above)
Comment (akheron) [reply frontend/src/employee-frontend/components/child-information/DailyServiceTimesSection.tsx]: The error message was not actually used anywhere, replaced with `true`

File: frontend/src/e2e-playwright/pages/employee/child-information.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    )
+  }
+
+  async openCollapsible(collapsible: Collapsible) {
+    const element = this.#collapsibles[collapsible]
+    await element.click()
+    return element
+  }
+
+  async openDailyServiceTimesCollapsible() {
```
Comment (Gekkio) [inline frontend/src/e2e-playwright/pages/employee/child-information.ts:22]: Aren't `openCollapsible` and `openDailyServiceTimesCollapsible` doing the same thing?

**Review by Joosakur — APPROVED** [review]
code looks good to me!

**Review by Gekkio — COMMENTED** [review]
The E2E test has one systematic problem: there are not waits, so all those `expect(...)` are evaluated once and if the page hasn't yet loaded something or otherwise the state hasn't yet stabilized, we get a flaky test.

**Comment by akheron** [comment]
> The E2E test has one systematic problem: there are not explicit waits, so all those `expect(...)` are evaluated once and if the page hasn't yet loaded something or otherwise the state hasn't yet stabilized, we get a flaky test.

Makes sense, fixed!

File: frontend/src/e2e-playwright/pages/employee/child-information.ts (hunk 4 above)
Comment (akheron) [reply frontend/src/e2e-playwright/pages/employee/child-information.ts]: Tweaked a bit, hope it's better now!

**Pull request merged**